File: Math/Python/fibonacci.py
def calcular_fibonacci():
    try:
        numero_str = input("Digite um número inteiro: ")
        numero = int(numero_str)

        if numero < 0:
            print("Entrada inválida. Por favor, digite um número inteiro não negativo.")
            return

        fibonacci = [0, 1]
        for i in range(2, numero + 1):
            fibonacci.append(fibonacci[i - 1] + fibonacci[i - 2])

        print(f"Sequência de Fibonacci até {numero}: {', '.join(map(str, fibonacci[:numero + 1]))}")

    except ValueError:
        print("Entrada inválida. Por favor, digite um número inteiro.")

calcular_fibonacci()

File: Math/Js/fibonacci.js
const prompt = require('prompt-sync')(); // Usar a função prompt() diretamente

function calcularFibonacciJS() {
    const input = prompt("Digite um número inteiro:");
    const numero = parseInt(input);
  
    if (isNaN(numero) || !Number.isInteger(numero) || numero < 0) {
      console.log("Entrada inválida. Por favor, digite um número inteiro não negativo.");
      return;
    }
  
    const fibonacci = [0, 1];
    for (let i = 2; i <= numero; i++) {
      fibonacci[i] = fibonacci[i - 1] + fibonacci[i - 2];
    }
  
    console.log(`Sequência de Fibonacci até ${numero}: ${fibonacci.slice(0, numero + 1).join(', ')}`);
  }
  
  calcularFibonacciJS();
  